From c3d9502059c773d3e693ffc3b12014c0c9233a16 Mon Sep 17 00:00:00 2001
From: Ian Stewart <istewart@nvidia.com>
Date: Tue, 30 Aug 2022 15:32:06 -0600
Subject: [PATCH] Disable outlining atomics

GCC 12 outlines atomics by default, but this is
not supported for the UEFI environment and so they
must remain inlined.
---
 cmake/AJACommonFlags.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/AJACommonFlags.cmake b/cmake/AJACommonFlags.cmake
index af8cbc1..b4677dc 100644
--- a/cmake/AJACommonFlags.cmake
+++ b/cmake/AJACommonFlags.cmake
@@ -11,8 +11,8 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL MSVC)
     endif()
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "^((Apple)Clang|GNU)$")
     if (CMAKE_SYSTEM_NAME MATCHES "^(Linux|Darwin)$")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC -mno-outline-atomics")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -mno-outline-atomics")
     endif()
     if(CMAKE_BUILD_TYPE MATCHES "^(Debug|RelWithDebInfo)$")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -O0 -g")
-- 
2.17.1

